### ISSUE TITLE
Build a classifier for dust features

### ISSUE THREAD

**Issue by dpalencia** [mars-vision/marsvision#3]
- [x] Check for overfitting in our cross validation grouping 
- [x] Plot ROC curves for our grouped data

Positive class: Dust
Negative class: No dust

Train the model Jupyter Notebooks initially, save to file for use in our pipeline.

Compare ROCs for different models.

We also want to convert the Jupyter Notebooks code into a training script.

**Comment by dpalencia** [comment]
Fixed up the DataLoader utility and used it to come up with a CSV of image features.

I'd still like to add on to the feature extractor to detect more features, though we can start doing some classification with this data.

**Comment by dpalencia** [comment]
We might want to shuffle the data as we train and test. Noticing that all of our rows with a dust class are together due to the way the data loader outputs to our CSV.

https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.ShuffleSplit.html#sklearn.model_selection.ShuffleSplit

**Comment by dpalencia** [comment]
Updated our sklearn notebook to use [GroupKFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GroupKFold.html#sklearn.model_selection.GroupKFold.split) to prevent overfitting on images in the same group.

The grouping was done by turning the region identifier digits for the HiRISE images into integers and using those as group labels. Doing this does seem to have brought the cross validation scores down, so it looks like this reduced the overfitting somewhat.

The ROC curve code in the notebook just needs to be updated to work with the grouped data now.

**Comment by dpalencia** [comment]
@mcorral98 
Let's double check our work with the GroupKFolds since we're still getting higher scores than expected on our cross validations.
